diff -urN faad2-2.7.org/frontend/main.c faad2-2.7/frontend/main.c
--- faad2-2.7.org/frontend/main.c	2008-09-22 19:55:09.000000000 +0200
+++ faad2-2.7/frontend/main.c	2016-10-28 22:19:51.733989150 +0200
@@ -31,7 +31,6 @@
 #ifdef _WIN32
 #define WIN32_LEAN_AND_MEAN
 #include <windows.h>
-#define off_t __int64
 #else
 #include <time.h>
 #endif
diff -urN faad2-2.7.org/libfaad/Makefile.am faad2-2.7/libfaad/Makefile.am
--- faad2-2.7.org/libfaad/Makefile.am	2009-02-05 01:51:03.000000000 +0100
+++ faad2-2.7/libfaad/Makefile.am	2016-10-28 22:19:57.386943593 +0200
@@ -4,7 +4,7 @@
 include_HEADERS = $(top_srcdir)/include/faad.h \
 		  $(top_srcdir)/include/neaacdec.h
 
-libfaad_la_LDFLAGS = -version-info 2:0:0
+libfaad_la_LDFLAGS = -version-info 2:0:0  -no-undefined
 libfaad_la_LIBADD = -lm
 
 libfaad_la_SOURCES = bits.c cfft.c decoder.c drc.c \
